File: backend/src/main/java/tom/projects/monkeylog/dto/workout/WorkoutResponse.java
package tom.projects.monkeylog.dto.workout;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import tom.projects.monkeylog.model.workout.WorkoutType;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class WorkoutResponse {
    @NotNull
    private UUID id;
    @NotEmpty
    private String name;
    @NotEmpty
    private String note;
    @NotNull
    private WorkoutType workoutType;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
}

File: backend/src/main/kotlin/tom/com/monkeylog/dto/workout/WorkoutResponse.kt
package tom.com.monkeylog.dto.workout

import tom.com.monkeylog.model.workout.WorkoutType
import java.time.LocalDateTime
import java.util.*

data class WorkoutResponse(
    val id: UUID,
    val name: String,
    val note: String? = null,
    val workoutType: WorkoutType,
    val startDate: LocalDateTime? = null,
    val endDate: LocalDateTime? = null,
)
